Shopify/cli
# Localhost GraphiQL

### WHY are these changes introduced?

Allows us to maintain query history and not lose work. Also, if `dev` dies or is killed, you can restart `dev` and your tab starts working again. Woot!


### WHAT is this pull request doing?


Uses a primary method of localhost access to GraphiQL, while leaving the proxy-based option open for now (though we should consider removing it entirely).

Defaults to port 3457 for serving; we should give the option of specifying another port by ENV variable and/or config file to enable running `dev` on multiple apps simultaneously.

### How to test your changes?


1. Spin up `dev` and verify that GraphiQL still works.
2. Quit `dev` and restart. Your tab should restore itself.
3. Open a new tab after the restart, and it should still keep your history.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
